PostHogTests/PostHogSurveysTest.swift

```diff
@@ -0,0 +1,201 @@
+        }
+
+        @Test("Survey decodes correctly")
+        func surveyDecodesCorrectly() throws {
+            let data = try loadFixture("fixture_survey_basic")
+            let sut = try PostHogApi.jsonDecoder.decode(Survey.self, from: data)
+
+            #expect(sut.id == "01947134-8a35-0000-549a-193b86fa2e44")
+            #expect(sut.name == "Core Web Vitals feature request")
+            #expect(sut.type == .popover)
+            #expect(sut.featureFlagKeys == [
+                SurveyFeatureFlagKeyValue(key: "flag1", value: "linked-flag-key"),
+                SurveyFeatureFlagKeyValue(key: "flag2", value: "survey-targeting-flag-key"),
+            ])
+            #expect(sut.targetingFlagKey == "survey-targeting-flag-key")
+            #expect(sut.internalTargetingFlagKey == "survey-targeting-88af4df282-custom")
+            #expect(sut.questions.count == 1)
+
+            if case let .basic(question) = sut.questions[0] {
+                #expect(question.question == "What would you like to see in our Core Web Vitals product?")
+                #expect(question.description == "Core Web Vitals just launched and we're interested in hearing (reading?) from you on what you think we're missing with this product")
+                #expect(question.originalQuestionIndex == 0)
+                #expect(question.descriptionContentType == .text)
+            } else {
+                throw TestError("Expected basic question type")
+            }
+
+            #expect(sut.conditions?.url == "core-web-vitals")
+            #expect(sut.appearance?.position == .right)
+            #expect(sut.appearance?.fontFamily == "system-ui")
+            #expect(sut.appearance?.whiteLabel == false)
+            #expect(sut.appearance?.borderColor == "#c9c6c6")
+            #expect(sut.appearance?.placeholder == "Start typing...")
+            #expect(sut.appearance?.backgroundColor == "#eeeded")
+            #expect(sut.appearance?.ratingButtonColor == "white")
+            #expect(sut.appearance?.submitButtonColor == "black")
+            #expect(sut.appearance?.submitButtonTextColor == "white")
+            #expect(sut.appearance?.thankYouMessageHeader == "Thank you for your feedback!")
+            #expect(sut.appearance?.displayThankYouMessage == true)
+            #expect(sut.appearance?.ratingButtonActiveColor == "black")
+            #expect(sut.appearance?.surveyPopupDelaySeconds == 5)
+            #expect(sut.startDate.map(ISO8601DateFormatter().string) == "2025-01-16T22:23:38Z")
+            #expect(sut.endDate == nil)
+            #expect(sut.currentIteration == nil)
+            #expect(sut.currentIterationStartDate == nil)
+        }
+
+        @Test("Survey question types decode correctly")
+        func surveyQuestionTypesDecodeCorrectly() throws {
+            let data = try loadFixture("fixture_survey_question_types")
+            let sut = try PostHogApi.jsonDecoder.decode(Survey.self, from: data)
+
+            // Basic question
```

Comment:
each question type should be its own test, so if it fails, we know each type failed